site/app/libraries/socket/Server.php

```diff
@@ -214,15 +214,20 @@ public function onMessage(ConnectionInterface $from, $msgString) {
         $msg = json_decode($msgString, true);
 
         if ($msg["type"] === "new_connection") {
-            if (!array_key_exists($msg['page'], $this->clients)) {
-                $this->clients[$msg['page']] = new \SplObjectStorage();
-            }
-            $this->clients[$msg['page']]->attach($from);
-            $this->setSocketClientPage($msg['page'], $from);
+            if (isset($msg['page'])) {
+                if (!array_key_exists($msg['page'], $this->clients)) {
+                    $this->clients[$msg['page']] = new \SplObjectStorage();
+                }
+                $this->clients[$msg['page']]->attach($from);
+                $this->setSocketClientPage($msg['page'], $from);
 
-            if ($this->core->getConfig()->isDebug()) {
-                $course_page = explode('-', $this->getSocketClientPage($from));
-                echo "New connection --> user_id: '" . $this->getSocketUserID($from) . "' - course: '" . $course_page[0] . "' - page: '" . $course_page[1] . "'\n";
+                if ($this->core->getConfig()->isDebug()) {
+                    $course_page = explode('-', $this->getSocketClientPage($from));
+                    echo "New connection --> user_id: '" . $this->getSocketUserID($from) . "' - course: '" . $course_page[0] . "' - page: '" . $course_page[1] . "'\n";
+                }
+            }
+            else {
+                $from->close();
             }
```

Comment:
Small recommendation here would be to log a message using the `Logger` class in `Logger.php` when this block is reached.  We shouldn't really end up here but if we do it would be nice to have something logged to help debugging by sys admins.